### ISSUE TITLE
Public Share No Viewer

### ISSUE THREAD

**Issue by gary-kim** [nextcloud/richdocuments#1199]
### Steps to reproduce
1. Share a PDF with a link
2. Open the share link

### Expected behaviour
Tell us what should happen

PDF viewer should show the PDF.

### Actual behaviour
Tell us what happens instead

Only a download link is shown.

It seems like there is supposed to be a preview image but that is not loading properly and I'd like there to be a full pdf viewer instead as well.

### Server configuration

**Operating system**: `Linux 5.4.0-48-generic`

**Web server:** `Apache`

**Database:** `mysql 8.0.21`

**PHP version:** `7.4.3`

**Nextcloud version:** (see Nextcloud admin page) `20.0.0, Gary Kim Personal Build - 20.0.0.9` (Also reproducible on download.nextcloud.com release)

**Where did you install Nextcloud from:** Personal build script

**List of activated apps:**

```
Enabled:
  - accessibility: 1.6.0
  - activity: 2.13.1
  - calendar: 2.1.2
  - cloud_federation_api: 1.3.0
  - comments: 1.10.0
  - contacts: 3.4.0
  - contactsinteraction: 1.1.0
  - cospend: 1.1.0
  - dashboard: 7.0.0
  - dav: 1.16.0
  - deck: 1.1.0
  - federatedfilesharing: 1.10.1
  - federation: 1.10.1
  - files: 1.15.0
  - files_external: 1.11.1
  - files_markdown: 2.3.1
  - files_pdfviewer: 2.0.1
  - files_rightclick: 0.17.0
  - files_sharing: 1.12.0
  - files_trashbin: 1.10.1
  - files_versions: 1.13.0
  - files_videoplayer: 1.9.0
  - firstrunwizard: 2.9.0
  - integration_github: 0.0.5
  - integration_moodle: 0.0.2
  - issuetemplate: 0.7.0
  - logreader: 2.5.0
  - lookup_server_connector: 1.8.0
  - maps: 0.1.8
  - metadata: 0.12.0
  - news: 15.0.2
  - nextcloud_announcements: 1.9.0
  - notes: 4.0.0
  - notifications: 2.8.0
  - oauth2: 1.8.0
  - password_policy: 1.10.1
  - photos: 1.2.0
  - privacy: 1.4.0
  - provisioning_api: 1.10.0
  - recommendations: 0.8.0
  - richdocuments: 3.7.4
  - riotchat: 0.6.11
  - serverinfo: 1.10.0
  - settings: 1.2.0
  - sharebymail: 1.10.0
  - sociallogin: 3.4.0
  - spreed: 10.0.0
  - systemtags: 1.10.0
  - talk_matterbridge: 1.18.3
  - tasks: 0.13.3
  - text: 3.1.0
  - theming: 1.11.0
  - twofactor_backupcodes: 1.9.0
  - twofactor_totp: 5.0.0
  - twofactor_u2f: 6.0.0
  - updatenotification: 1.10.0
  - user_status: 1.0.0
  - viewer: 1.4.0
  - weather_status: 1.0.0
  - workflowengine: 2.2.0
```

### Client configuration
**Browser:** Firefox 

**Operating system:** Linux

**Comment by skjnldsv** [comment]
For you @juliushaertl (he's in vacations, will have to wait)

**Comment by gary-kim** [comment]
Okay, I think I narrowed it down. 

`application/pdf` is part of `OC.getCapabilities().richdocuments.mimetypesNoDefaultOpen` so by default, the viewer will not open PDFs. It does open it if download is hidden because then `OC.getCapabilities().richdocuments.mimetypesNoDefaultOpen` is ignored.

https://github.com/nextcloud/richdocuments/blob/886ddb62ce5fbcb8ba210ce90c27a78066b3188a/src/files.js#L268-269

**Comment by skjnldsv** [comment]
> @juliushaertl Could we add a special case for `application/pdf` on public viewer pages so the no default open is ignored? We still don't want it to open PDFs by default usually.

if that is so, I can just make the pdfviewer by default.
Who wants to open collabora in those cases?
cc @jancborchardt

**Comment by gary-kim** [comment]
Well, now I'm confused. Just tried `hide download` to see if `richdocuments` opens properly and funny enough, `files_pdfviewer` came up instead of `richdocuments`. Looks like `files_pdfviewier` will show on hidden download because `canDownload` will return `false` which means that `isSecureViewerAvailable()` is now false.

**Comment by gary-kim** [comment]
Different topic of discussion but I think it makes sense to have LOOL open PDF files. Especially for big ones where it can take a while to download the entire file locally, it might be faster to use LOOL since it probably has a faster connection and it only needs to send images of the PDF rather than the entire file. Again, different topic of discussion. We can make it another issue.

**Comment by joshtrichards** [comment]
This appears to have been covered in a later issue #1320 and fixed in nextcloud/files_pdfviewer#292.

Also confirmed correct behavior today w/ and w/o *Hide download* set in NC27:

- *Hide download* unset = uses files_pdfviewer
- "Hide download* set = loads in CODE in SecureView mode
